trulens/nn/models/_model_base.py

```diff
@@ -172,8 +242,24 @@ def fprop(
                 as the input. If `attribution_cut` is supplied, also return the
                 cut activations.
         """
+
+        # TODO: Can attribution_cut preceed doi_cut or follow to_cut in the model?
```

Comment:
we can remove this todo and add to documentation. There are 2 questions here:
1) attribution cut cannot preceed  doi_cut (or gradients would break since expansion is not differentiable)
2) to_cut is not specifically tied to any of the attribution progression, it just says to which layer the fprop is going to. so it can be before or after any cut, as it can be to any layer.

Reply:
Moving the documentation about these restrictions to bprop as I don't think they matter for fprop; fprop doesn't take any gradients

Reply:
Also I'm unsure about your point (2), looks like the gradients done in bprop are to_cut w.r.t. attribution_cut so that implies an order regarding these two which would produce errors were it not to hold.

On the other hand, if attribution was before doi_cut, the evaluation might run but have unexpected results.